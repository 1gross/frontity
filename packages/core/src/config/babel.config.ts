import { TransformOptions } from "babel-core";
import { Env, Mode, BabelConfigs } from "./types";

const targets = {
  module: { esmodules: true },
  es5: {
    browsers: [
      "and_chr >= 67",
      "and_ff >= 18",
      "and_uc >= 11.8",
      "android >= 67",
      "not android <= 4.4.4",
      "chrome >= 49",
      "edge >= 12",
      "firefox >= 18",
      "ios_saf >= 10",
      "not op_mini all",
      "op_mob >= 46",
      "opera >= 36",
      "safari >= 10",
      "samsung >= 5"
    ]
  },
  node: { node: "8.10" }
};

export default (mode: Mode): BabelConfigs => {
  const getConfig = (env: Env): TransformOptions => {
    const presets = [
      "@babel/preset-typescript",
      ["@babel/preset-env", { useBuiltIns: "usage", targets: targets[env] }],
      "@babel/preset-react"
    ];
    const plugins = [
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties"
    ];
    if (mode === "development" && env === "module")
      plugins.push("react-hot-loader/babel");
    return {
      presets,
      plugins
    };
  };

  return {
    module: getConfig("module"),
    es5: getConfig("es5"),
    node: getConfig("node")
  };
};
