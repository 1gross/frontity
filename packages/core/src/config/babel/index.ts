import { TransformOptions } from "babel-core";
import { Target, Mode, BabelConfigs } from "../../types";

const targets = {
  // Browsers with <script type="module"></script> support.
  module: { esmodules: true },
  // Browsers with Proxy support, which is needed by Overmind.
  // For older browsers (ie8-11) we support AMP fallback.
  es5: {
    browsers: [
      "and_chr >= 67",
      "and_ff >= 18",
      "and_uc >= 11.8",
      "android >= 67",
      "not android <= 4.4.4",
      "chrome >= 49",
      "edge >= 12",
      "firefox >= 18",
      "ios_saf >= 10",
      "not op_mini all",
      "op_mob >= 46",
      "opera >= 36",
      "safari >= 10",
      "samsung >= 5"
    ]
  },
  // Node version used by AWS Lambda.
  node: { node: "8.10" }
};

export default ({ mode }: { mode: Mode }): BabelConfigs => {
  const getConfig = (target: Target): TransformOptions => {
    const presets = [
      // Instead of using a TS transpiler, this removes the typescript code.
      "@babel/preset-typescript",
      [
        "@babel/preset-env",
        {
          targets: targets[target]
          // esModules and CJS don't work well together so this is here
          // to be able to parse node_modules, where there are many modules
          // still using CJS. It will probably break tree shaking, so we need
          // to check that later. Another option would be the opposite:
          // https://www.npmjs.com/package/babel-plugin-transform-commonjs-es2015-modules
          // modules: target !== "node" ? "false" : "commonjs"
        }
      ],
      "@babel/preset-react"
    ];
    const plugins = [
      // Support for dynamic imports: import("./my-file")
      "@babel/plugin-syntax-dynamic-import",
      // Needed for loadable-component SSR.
      "@loadable/babel-plugin",
      // Support for the rest spread: { ...obj }
      "@babel/plugin-proposal-object-rest-spread",
      // Support for the class props: class MyClass { myProp = 'hi there' }
      "@babel/plugin-proposal-class-properties"
    ];
    return {
      presets,
      plugins
    };
  };

  return {
    module: getConfig("module"),
    es5: getConfig("es5"),
    node: getConfig("node")
  };
};
