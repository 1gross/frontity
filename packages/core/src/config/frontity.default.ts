import FrontityConfig from '.';

const defaultConfig: FrontityConfig = {
  headers: {
    'Access-Control-Allow-Origin': '*',
    'Origin, X-Requested-With, Content-Type': 'Accept',
  },
  staticHeaders: {},
  dynamicHeaders: {},
  robots: 'User-agent: *\nDisallow:',
};

export default defaultConfig;
