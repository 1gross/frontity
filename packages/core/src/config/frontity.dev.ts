import FrontityConfig from '.';

const config = (config: FrontityConfig) => {
  config.staticHeaders['Cache-Control'] = 'no-cache, no-store, must-revalidate';
  config.dynamicHeaders['Cache-Control'] =
    'no-cache, no-store, must-revalidate';
};

export default config;
