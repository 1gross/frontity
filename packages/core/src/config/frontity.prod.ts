import FrontityConfig from '.';

const config = (config: FrontityConfig) => {
  config.staticHeaders['Cache-Control'] =
    'public, max-age=31536000, s-maxage=31536000, immutable';
  config.dynamicHeaders['Cache-Control'] =
    'public, max-age=0, s-maxage=120, stale-while-revalidate=31536000, stale-if-error=31536000';
};

export default config;
