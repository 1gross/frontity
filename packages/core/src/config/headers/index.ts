import { Mode, HeaderConfig } from "../../types";

export default ({ mode }: { mode: Mode }): HeaderConfig => {
  const commonHeaders = {
    "Access-Control-Allow-Origin": "*",
    "Origin, X-Requested-With, Content-Type": "Accept"
  };
  const staticHeaders = {};
  const dynamicHeaders = {};
  if (mode === "development") {
    commonHeaders["Cache-Control"] = "no-cache, no-store, must-revalidate";
  } else {
    staticHeaders["Cache-Control"] =
      "public, max-age=31536000, s-maxage=31536000, immutable";
    dynamicHeaders["Cache-Control"] =
      "public, max-age=0, s-maxage=120, stale-while-revalidate=31536000, stale-if-error=31536000";
  }
  return {
    static: { ...staticHeaders, ...commonHeaders },
    dynamic: { ...dynamicHeaders, ...commonHeaders }
  };
};
