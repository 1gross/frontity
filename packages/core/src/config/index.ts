import { Mode, FrontityConfig } from "./types";
import getHeaders from "./headers";
import getBabel from "./babel";
import getWebpack from "./webpack";
import getRobots from "./robots";

export default ({ mode }: { mode: Mode }): FrontityConfig => {
  const headers = getHeaders({ mode });
  const babel = getBabel({ mode });
  const webpack = getWebpack({ mode, babel });
  const robots = getRobots({ mode });
  return {
    headers,
    babel,
    webpack,
    robots
  };
};
