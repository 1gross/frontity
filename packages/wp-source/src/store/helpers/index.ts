export const getIdBySlug = (
  entityMap: {
    [id: number]: {
      slug: string;
      id: number;
    };
  },
  slug: string
): number | undefined => {
  const entity = Object.values(entityMap).find(e => e.slug === slug);
  return entity && entity.id;
};

export const getTotal = (response: Response): number =>
  parseInt(response.headers.get("X-WP-Total"));

export const getTotalPages = (response: Response): number =>
  parseInt(response.headers.get("X-WP-TotalPages"));

export const normalizePath = (pathOrLink: string): string => {
  try {
    return new URL(pathOrLink).pathname;
  } catch (e) {
    // is not a URL
    return pathOrLink;
  }
};
