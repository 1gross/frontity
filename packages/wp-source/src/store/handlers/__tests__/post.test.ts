import { createOvermindMock } from "overmind";
import { namespaced } from "overmind/config";
import { state, actions, effects } from "../../";
import handler from "../post";

import { mockResponse } from "./mocks/helpers";

const post60 = {
  id: 60,
  date: "2016-11-25T18:31:11",
  slug: "the-beauties-of-gullfoss",
  type: "post",
  link: "https://test.frontity.io/2016/the-beauties-of-gullfoss/",
  title: {},
  content: {},
  excerpt: {},
  author: 4,
  featured_media: 62,
  meta: [],
  categories: [57, 59, 56, 3, 8, 58],
  tags: [10, 9, 13, 11]
};

let mocks;
beforeEach(() => {
  mocks = {
    // Mock resolver
    resolver: {
      match: jest.fn().mockReturnValue({
        handler,
        params: { slug: "the-beauties-of-gullfoss" }
      })
    },
    // Mock api
    api: {
      get: jest.fn().mockResolvedValue(mockResponse([post60]))
    },
    // Populate
    populate: jest.fn().mockResolvedValue([
      {
        id: 60,
        slug: "the-beauties-of-gullfoss",
        link: "https://test.frontity.io/2016/the-beauties-of-gullfoss/"
      }
    ])
  };
});

describe("post", () => {
  test("doesn't exist in source.post", async () => {
    const config = namespaced({ source: { actions, state, effects } });
    const store = createOvermindMock(config, { source: mocks });

    await store.actions.source.fetch("/the-beauties-of-gullfoss/");

    expect(store.state.source.dataMap).toMatchSnapshot();
  });

  test("exists in source.post", async () => {
    const config = namespaced({
      source: {
        actions,
        state: {
          ...state,
          post: { 60: post60 }
        },
        effects
      }
    });
    const store = createOvermindMock(config, { source: mocks });

    await store.actions.source.fetch("/the-beauties-of-gullfoss/");

    expect(mocks.api.get).not.toBeCalled();
    expect(store.state.source.dataMap).toMatchSnapshot();
  });

  test("throws an error if it doesn't exist", async () => {
    mocks.api.get = jest.fn().mockResolvedValueOnce(mockResponse([]));
    mocks.populate = jest.fn().mockResolvedValueOnce(mockResponse([]));

    const config = namespaced({ source: { actions, state, effects } });
    const store = createOvermindMock(config, { source: mocks });

    await store.actions.source.fetch("/the-beauties-of-gullfoss/");

    expect(store.state.source.dataMap).toMatchSnapshot();
  });
});
