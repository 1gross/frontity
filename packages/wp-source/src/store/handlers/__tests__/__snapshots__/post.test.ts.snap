// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post doesn't exist in source.post 1`] = `
Object {
  "/the-beauties-of-gullfoss/": Object {
    "id": 60,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
}
`;

exports[`post doesn't exist in source.post 2`] = `
Object {
  "attachment": Array [
    13,
    62,
    70,
    84,
    95,
    118,
  ],
  "author": Array [
    4,
  ],
  "category": Array [
    3,
    8,
    56,
    57,
    58,
    59,
  ],
  "post": Array [
    60,
  ],
  "tag": Array [
    9,
    10,
    11,
    13,
  ],
}
`;

exports[`post exists in source.post 1`] = `
Object {
  "/the-beauties-of-gullfoss/": Object {
    "id": 60,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
}
`;

exports[`post exists in source.post 2`] = `
Object {
  "attachment": Array [],
  "author": Array [],
  "category": Array [],
  "post": Array [
    60,
  ],
  "tag": Array [],
}
`;

exports[`post throws an error if it doesn't exist 1`] = `
Object {
  "/the-beauties-of-gullfoss/": Object {
    "is404": true,
    "isFetching": false,
  },
}
`;

exports[`post throws an error if it doesn't exist 2`] = `
Object {
  "attachment": Array [],
  "author": Array [],
  "category": Array [],
  "post": Array [],
  "tag": Array [],
}
`;
